# 01 importing libraries

In [2]:
import pandas as pd
import numpy as np
import os

# 02 importing each data set

In [3]:
ords_prods_combined = pd.read_pickle('orders_products_combined.pkl')

In [4]:
ords_prods_combined.shape

(32434489, 11)

In [5]:
ords_prods_combined.head()

order_id  user_id eval_set  order_number  orders_day_of_week  \
0   2539329        1    prior             1                   2   
1   2539329        1    prior             1                   2   
2   2539329        1    prior             1                   2   
3   2539329        1    prior             1                   2   
4   2539329        1    prior             1                   2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     NaN         196                  1   
1                  8                     NaN       14084                  2   
2                  8                     NaN       12427                  3   
3                  8                     NaN       26088                  4   
4                  8                     NaN       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

#data set ords_prods_combined is the same size as exported already:32434489, 11

In [7]:
prods_checked = pd.read_csv('products_checked.csv', index_col= 0)

In [8]:
prods_checked.head()

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [9]:
df_merged= ords_prods_combined.merge(prods_checked, on='product_id', indicator='merge_source')

In [12]:
df_merged['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

In [10]:
df_merged.shape

(32404859, 16)

In [11]:
df_merged.head()

order_id  user_id eval_set  order_number  orders_day_of_week  \
0   2539329        1    prior             1                   2   
1   2539329        1    prior             1                   2   
2   2539329        1    prior             1                   2   
3   2539329        1    prior             1                   2   
4   2539329        1    prior             1                   2   

   order_hour_of_day  days_since_prior_order  product_id  add_to_cart_order  \
0                  8                     NaN         196                  1   
1                  8                     NaN       14084                  2   
2                  8                     NaN       12427                  3   
3                  8                     NaN       26088                  4   
4                  8                     NaN       26405                  5   

   reordered _merge                             product_name  aisle_id  \
0          0   both                                     Soda        77   
1          0   both  Organic Unsweetened Vanilla Almond Milk        91   
2          0   both                      Original Beef Jerky        23   
3          0   both               Aged White Cheddar Popcorn        23   
4          0   both         XL Pick-A-Size Paper Towel Rolls        54   

   department_id  prices merge_source  
0              7     9.0         both  
1             16    12.5         both  
2             19     4.4         both  
3             19     4.7         both  
4             17     1.0         both

In [ ]:
# df_merged2= ords_prods_combined.merge(prods_checked, on='product_id',how= 'outer', indicator='merge_source')
# I could not run this line of code, despite I checked the availability of enough space in my system and everything was OK, 
# this line of code was broke several times and never ran correctly.

In [13]:
df_merged.to_pickle('ords_prods_merge.pkl')